plugins/decorators/legacy-doc-url-decorator.js

```diff
@@ -0,0 +1,50 @@
+const X_LEGACY_DOC_URLS_PROPERTY = "x-legacy-doc-urls";
+
+/**
+ * The decorator prepends HTML anchors to the description of each operation with HTML fragments leading to related
+ * legacy documentation pages. This achieves link backward compatibility.
+ *
+ * For example, if the old URL is https://docs.apify.com/api/v2#/reference/logs/log/get-log, then we prepend:
+ *   <span id="/reference/logs/log/get-log"></span>
+ *
+ * The legacy URLs are stored on a custom property `x-legacy-doc-urls` in the node object (typically an operation
+ * or tag). Multiple URLs per node are supported (we might point multiple URLs to the same tag or operation).
+ */
+function prependLegacyUrlAnchor(target) {
+    if (X_LEGACY_DOC_URLS_PROPERTY in target) {
+        const debugId = target.operationId || target.name || target.summary;
+
+        if (!Array.isArray(target[X_LEGACY_DOC_URLS_PROPERTY])) {
+            console.warn(`Invalid legacy doc URL on ${debugId}. Expected non-empty array`);
+            return;
+        }
+
+        const anchors = target[X_LEGACY_DOC_URLS_PROPERTY]
+            .map(url => {
+                const [_, fragment] = url.split("#");
+
+                if (!fragment) {
+                    console.warn(`Invalid legacy doc URL on ${debugId}: ${url}`);
+                }
+
+                return fragment;
+            })
+            .filter(fragment => fragment)
```

Comment:
Isn't this line redundant? I'm wondering if it's doing anything at all.

Reply:
If you have some invalid URLs, this will skip them, preventing empty `<span />` being added.

Reply:
Understood.

```
> '' == false
true
```